### DISCUSSION TITLE
There is insufficient memory for the Java Runtime Environment to continue

### DISCUSSION THREAD

**Discussion by jim1991** [apache/skywalking#9451] [Q&A]
I run this application with docker images and I get this problem, How can I solve it ?

[0.002s][warning][os,thread] Failed to start thread - pthread_create failed (EPERM) for attributes: stacksize: 1024k, guardsize: 4k, detached.

There is insufficient memory for the Java Runtime Environment to continue.
Cannot create worker GC thread. Out of system resources.
Can not save log file, dump to screen..
#
There is insufficient memory for the Java Runtime Environment to continue.
Cannot create worker GC thread. Out of system resources.
 Possible reasons:
The system is out of physical RAM or swap space
The process is running with CompressedOops enabled, and the Java Heap may be blocking the growth of the native heap

**Comment by wu-sheng** [comment]
Why should this question be answered by the community?
This is an OS resource or docker setup issue, basically.

**Comment by jaychoww** [comment]
I had the same problem, I couldn't bootstrup the skywalking later than 9.0.0 via docker-compose. It reported the same message(memory insufficient).
My solution is to upgrade the docker to the latest verion(now is Docker version 20.10.18, build b40c2f6), and it works now.
@jim1991

**Reply by jim1991** [reply]
这是Java11 在 docker里面的安全配置问题，使用docker启动时需要添加  --security-opt=seccomp:unconfined，使用docker-compose时需要添加
```yaml
      image: xxxx:xx
      security_opt:
      - seccomp:unconfined
```
